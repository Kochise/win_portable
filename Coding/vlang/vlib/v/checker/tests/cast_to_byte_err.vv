type SAlias = string

type CAlias = char
type IAlias = int
type UAlias = u32
type FAlias = f32

fn main() {
	// should be errors:
	_ := byte('hello')
	_ := byte(SAlias('hello'))

	// should be allowed:
	_ := byte(char(1))
	_ := byte(int(1))
	_ := byte(u32(1))
	_ := byte(f32(1.0))

	_ := byte(CAlias(1))
	_ := byte(IAlias(1))
	_ := byte(UAlias(1))
	_ := byte(FAlias(1))
}
