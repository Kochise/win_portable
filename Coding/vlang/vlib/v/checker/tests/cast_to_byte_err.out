vlib/v/checker/tests/cast_to_byte_err.vv:10:7: error: cannot cast type `string` to `byte`
    8 | fn main() {
    9 |     // should be errors:
   10 |     _ := byte('hello')
      |          ~~~~~~~~~~~~~
   11 |     _ := byte(SAlias('hello'))
   12 |
vlib/v/checker/tests/cast_to_byte_err.vv:11:7: error: cannot cast type `SAlias` to `byte`
    9 |     // should be errors:
   10 |     _ := byte('hello')
   11 |     _ := byte(SAlias('hello'))
      |          ~~~~~~~~~~~~~~~~~~~~~
   12 |
   13 |     // should be allowed:
